app/custom-sounds/client/admin/EditSound.js

```diff
@@ -0,0 +1,238 @@
+
+	const validate = (soundData, soundFile) => {
+		const errors = [];
+
+		if (!soundData._id) {
+			errors.push('_id');
+		}
+
+		if (!soundData.name) {
+			errors.push('Name');
+		}
+
+		errors.forEach((error) => dispatchToastMessage({ type: 'error', message: error }));
+		if (soundFile !== previousSound) {
+			if (!/audio\/mp3/.test(soundFile.type) && !/audio\/mpeg/.test(soundFile.type) && !/audio\/x-mpeg/.test(soundFile.type)) {
+				errors.push('FileType');
+				dispatchToastMessage({ type: 'error', message: t('error-invalid-file-type') });
+			}
+		}
+
+		return errors.length === 0;
+	};
+
+	const saveAction = async (sound) => {
+		const soundData = getSoundData(sound);
+		if (validate(soundData, sound)) {
+			let soundId;
+			try {
+				soundId = await insertOrUpdateSound(soundData);
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: error });
+			}
+
+			soundData._id = soundId;
+			soundData.random = Math.round(Math.random() * 1000);
+
+			if (sound && sound !== previousSound) {
+				dispatchToastMessage({ type: 'success', message: t('Uploading_file') });
+
+				const reader = new FileReader();
+				reader.readAsBinaryString(sound.soundFile);
+				reader.onloadend = () => {
+					console.log(reader.result, sound.soundFile.type, soundData);
+
+					try {
+						uploadCustomSound(reader.result, sound.soundFile.type, soundData);
+					} catch (error) {
+						dispatchToastMessage({ type: 'error', message: error });
+					}
+				};
+			}
+			return dispatchToastMessage({ type: 'success', message: t('File_uploaded') });
+		}
+	};
+
+	const handleSave = useCallback(async () => {
+		saveAction(sound);
+		onChange();
+	}, [name, _id, sound]);
+
+	const onDeleteConfirm = useCallback(async () => {
+		try {
+			await deleteCustomSound(_id);
+			console.log();
+			setModal(() => <SuccessModal onClose={() => { setModal(undefined); close(); onChange(); }}/>);
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+			onChange();
+		}
+	}, [_id]);
+
+	const openConfirmDelete = () => setModal(() => <DeleteWarningModal onDelete={onDeleteConfirm} onCancel={() => setModal(undefined)}/>);
+
+
+	const clickUpload = useFileInput(handleChangeFile, 'audio/mp3');
+
+
+	return <>
+		<Box display='flex' flexDirection='column' fontScale='p1' color='default' mbs='x20' {...props}>
+			<Margins block='x4'>
+				<Field>
+					<Field.Label>{t('Name')}</Field.Label>
+					<Field.Row>
+						<TextInput value={name} onChange={(e) => setName(e.currentTarget.value)} placeholder={t('Name')} />
+					</Field.Row>
+				</Field>
+
+				<Field>
+					<Field.Label alignSelf='stretch'>{t('Sound_File_mp3')}</Field.Label>
+					<Box display='flex' flexDirection='row' mbs='none'>
+						<Margins inline='x4'>
+							<Button square onClick={clickUpload}><Icon name='upload' size='x20'/></Button>
+							{(sound && sound.name) || 'none'}
```

Comment:
Need a better element to add this info, any suggestions?

Reply:
`<Box is='span' />`?